pytorch_lightning/__init__.py

```diff
@@ -76,6 +67,33 @@
         'seed_everything',
         'metrics',
     ]
+else:
+    try:
+        # This variable is injected in the __builtins__ by the build
+        # process. It used to enable importing subpackages of skimage when
+        # the binaries are not built
+        _ = None if __LIGHTNING_SETUP__ else None
+    except NameError:
+        __LIGHTNING_SETUP__: bool = False
+
+    if __LIGHTNING_SETUP__:  # pragma: no-cover
```

Comment:
```suggestion
    if not typing.TYPE_CHECKING or __LIGHTNING_SETUP__:  # pragma: no-cover
```
can we write it this way just to not maintain double imports?

Reply:
Haha! Yeah, had the same intuition as you here :( Sadly, it seems the "or" logic is a bit too complex for the type-checker, so it doesn't work.

My understanding is that `typing.TYPE_CHECKING` actually just gets statically replaced with the literal `False` or `True` before type-checking happens. So from the type-checker's point of view, the the suggested change becomes:

```
if True or __LIGHTNING_SETUP__:
   ...
```
so the imports are still conditional.

Reply:
Ah, yes, here's the explanation of how `TYPE_CHECKING` works: https://github.com/python/typing/issues/230

Reply:
I am sorry I could see it in the linked typing issue